pydust/src/builtins.zig

```diff
@@ -71,6 +71,46 @@ pub fn callable(object: anytype) bool {
     return ffi.PyCallable_Check(obj.py) == 1;
 }
 
+/// Call a callable object with no arguments.
+pub fn call0(comptime T: type, object: anytype) !T {
+    const result = ffi.PyObject_CallNoArgs(py.object(object).py) orelse return PyError.PyRaised;
+    return try py.as(T, result);
+}
+
+/// Call a callable object with the given arguments.
+pub fn call(comptime ReturnType: type, object: anytype, args: anytype, kwargs: anytype) !ReturnType {
+    const pyobj = py.object(object);
+
+    var argsPy: py.PyTuple = undefined;
+    if (@typeInfo(@TypeOf(args)) == .Optional and args == null) {
+        argsPy = try py.PyTuple.new(0);
+    } else {
+        argsPy = try py.PyTuple.checked(try py.create(args));
+    }
+    defer argsPy.decref();
+
+    // TODO(ngates): avoid creating empty dict for kwargs
```

Comment:
it's valid to pass NULL for kwargs in `PyObject_Call`

Reply:
Yeah I know, we just don't atm